dao/src/integ-test/java/uk/gov/moj/sdt/dao/BulkCustomerDaoTest.java

```diff
@@ -1,83 +1,177 @@
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.moj.sdt.dao.api.IBulkCustomerDao;
+import uk.gov.moj.sdt.dao.config.DaoTestConfig;
+import uk.gov.moj.sdt.domain.BulkCustomer;
+import uk.gov.moj.sdt.domain.api.IBulkCustomer;
+import uk.gov.moj.sdt.test.utils.AbstractIntegrationTest;
+import uk.gov.moj.sdt.test.utils.TestConfig;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+/**
+ * Test {@link BulkCustomerDao} query methods.
+ *
+ * @author Robin Compston
+ */
+@ActiveProfiles("integ")
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = { TestConfig.class, DaoTestConfig.class})
+@Sql(scripts = {"classpath:uk/gov/moj/sdt/dao/sql/BulkCustomerDaoTest.sql"})
+@Transactional
+public class BulkCustomerDaoTest extends AbstractIntegrationTest {
+    /**
+     * Logger object.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(BulkCustomerDaoTest.class);
+    CriteriaBuilder criteriaBuilder;
+    CriteriaQuery<BulkCustomer> criteriaQuery;
+    Root<BulkCustomer> root;
+    @Before
+    public void setup() {
+        final IBulkCustomerDao bulkCustomersDao = this.applicationContext.getBean(IBulkCustomerDao.class);
+        criteriaBuilder = bulkCustomersDao.getEntityManager().getCriteriaBuilder();
+        criteriaQuery = criteriaBuilder.createQuery(BulkCustomer.class);
+        root = criteriaQuery.from(BulkCustomer.class);
+    }
+
+    /**
+     * Tests {@link uk.gov.moj.sdt.dao.GenericDao} fetch.
+     */
+    @Test
+    public void testGetBulkCustomerBySdtId() {
+        final IBulkCustomerDao bulkCustomersDao = this.applicationContext.getBean(IBulkCustomerDao.class);
+
+        final IBulkCustomer bulkCustomer = bulkCustomersDao.getBulkCustomerBySdtId(2);
+        if (bulkCustomer != null) {
+            LOGGER.debug("Retrieved bulk customer id [" + bulkCustomer.getId() + "]");
+        } else {
+            Assert.fail("Could not find bulk customer [" + 2 + "]");
+        }
+    }
+
+    @Test
+    public void testFetchBulkCustomerId() {
+        final IBulkCustomerDao bulkCustomersDao = this.applicationContext.getBean(IBulkCustomerDao.class);
+
+        final IBulkCustomer bulkCustomer = bulkCustomersDao.fetch(IBulkCustomer.class, 10711);
+        if (bulkCustomer != null) {
+            LOGGER.debug("Retrieved bulk customer id [" + bulkCustomer.getId() + "]");
+        } else {
+            Assert.fail("Could not find bulk customer [" + 2 + "]");
```

Comment:
The 2 in the fail message should be 10711 (the customer id being searched for).

Reply:
done